FUEL/household.py

```diff
@@ -0,0 +1,255 @@
+        if type(primary_threshold) != int or primary_threshold < 0:
+            raise ValueError("The primary threshold must be a positive integer!")
+
+        stove_type = 0
+        if stove == "All":
+            stove_type = self.stoves
+        else:
+            for s in self.stoves:
+                if stove in s:
+                    stove_type = [s]
+        if stove_type == 0:
+            raise ValueError('Stove not found in data set.')
+
+        number_of_cooking_events = {}
+        cook_events_list = {}
+
+        for s in stove_type:
+            peaks = find_peaks(self.df_stoves[s].values, height=primary_threshold, distance=time_between_events)[0]
+            number_of_cooking_events.update({s.split(' ')[0]: len(peaks)})
+            cook_events_list.update({s: peaks})
+        self.cook_events = cook_events_list
+
+        return number_of_cooking_events
+
+    def plot_cooking_events(self, primary_threshold = 15, time_between_events = 60, stove="All"):
+        '''Must add .show() after calling function for plot to be generated'''
+
+        # The primary threshold must an integer >= 0
+        # The time between cooking events must be an integer >= 0
+        # The stove type must be entered as a string and must be in the data set
+        # If the primary threshold and time between events is not specified it will revert back to default value even
+        # if it was set to different values when calling cooking_events() function.
+
+        if type(stove) != str:
+            raise ValueError('Must input fuel type as a string!')
+        if type(time_between_events) != int or time_between_events < 0:
+            raise ValueError("The time between events must be a positive integer!")
+        if type(primary_threshold) != int or primary_threshold < 0:
+            raise ValueError("The primary threshold must be a positive integer!")
+
+        stove_type = 0
+        if stove == "All":
+            stove_type = self.stoves
+        else:
+            for s in self.stoves:
+                if stove in s:
+                    stove_type = [s]
+            if stove_type == 0:
+                raise ValueError('Stove not found in data set.')
+
+        self.cooking_events(primary_threshold, time_between_events, stove)
+        events = self.cook_events
+        fig = self.plot_stove(stove)
+
+        for s in stove_type:
+            fig.add_trace(
+                go.Scatter(x=self.df_stoves['Timestamp'][events[s]],
+                           y=self.df_stoves[s][events[s]],
+                           mode='markers',
+                           name=s.split(' ')[0] + 'Cooking Events'
+                           )
+                        )
+        return fig
+
+
+if __name__ == "__main__":
+
+    # x = Household('./data_files/HH_38_2018-08-26_15-01-40_processed_v3.csv')
+
+    # print(x.stove_types())
+    # x.plot_cooking_events('telia')
+
+    data_1 = Household('./data_files/test_4.csv')
```

Comment:
I suggest using filedialog for inputting file names:
"from tkinter import filedialog"
then use the askopenfilename() command...
"data_1=filedialog.askopenfilename()"
...this will primpt a file explorer window, and the name of the selected file is returned as a string